### PR TITLE
Ariba e2e changes

### PR DESCRIPTION
**Pull request by Vipinofficial11**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,6 +60,11 @@
 
   <dependencies>
 
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${guava.version}</version>
+    </dependency>
```
Comment (itsankit-google) [inline pom.xml:67]: nit - revert these changes

File: pom.xml [hunk 2]
```diff
@@ -271,18 +279,11 @@
       <scope>provided</scope>
     </dependency>
 
-    <dependency>
-      <groupId>com.google.guava</groupId>
-      <artifactId>guava</artifactId>
-      <version>${guava.version}</version>
-    </dependency>
-
```
Comment (itsankit-google) [inline pom.xml:279]: nit - revert these changes

File: pom.xml [hunk 3]
```diff
@@ -271,18 +279,11 @@
...
     <dependency>
       <groupId>com.google.code.gson</groupId>
       <artifactId>gson</artifactId>
       <version>${gson.version}</version>
     </dependency>
-
```
Comment (itsankit-google) [inline pom.xml]: nit - revert

File: src/e2e-test/features/aribasource/RunTime.feature [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+# the License.
+
+
+@AribaSource
+@Smoke
+@Regression
+Feature: Ariba Source - Run time scenarios
+
+  @BATCH-TS-ARIBA-DSGN-RNTM-01 @BQ_SINK @FILE_PATH @BQ_SINK_CLEANUP @BQ_TEMP_CLEANUP
+  Scenario: Verify user should be able to deploy the pipeline when plugin is configured for System Type prod
```
Comment (itsankit-google) [inline src/e2e-test/features/aribasource/RunTime.feature]: nit - `Verify data is transferred correctly to BQ for System Type prod`

File: src/e2e-test/features/aribasource/RunTimeWithMacro.feature [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+# the License.
+
+
+@AribaSource
+@Smoke
+@Regression
+Feature: Ariba Source - Run time scenarios (macros)
+
+  @BATCH-TS-ARIBA-DSGN-RNTM-MACRO-01 @BQ_SINK @FILE_PATH @BQ_SINK_CLEANUP @BQ_TEMP_CLEANUP
+  Scenario: Verify user should be able to deploy the pipeline when plugin is configured for System Type prod with macros
```
Comment (itsankit-google) [inline src/e2e-test/features/aribasource/RunTimeWithMacro.feature]: `Verify data is transferred correctly to BQ for System Type prod with macros`

File: src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+package io.cdap.plugin.aribasource.stepsdesign;
+import io.cdap.plugin.aribasource.actions.AribaSourcePropertiesPageActions;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import java.io.IOException;
+
+/**
+ * Represents - Ariba - Source plugin - Properties page - Steps.
+ */
+public class DesignTimeSteps {
```
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java]: Rename class, these steps are not design time.

File: src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ * the License.
+ */
+package io.cdap.plugin.aribasource.stepsdesign;
+import io.cdap.plugin.aribasource.actions.AribaSourcePropertiesPageActions;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import java.io.IOException;
+
+/**
+ * Represents - Ariba - Source plugin - Properties page - Steps.
```
Comment (itsankit-google) [inline src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java]: incorrect java doc

**Review by itsankit-google — COMMENTED** [review]
There is only one success scenario with data validation with system type `prod`. Do we expect to add other scenarios in future PRs??

File: pom.xml (hunk 1 above)
Comment (Vipinofficial11) [reply pom.xml:67]: This is actually done intentionally to override the other guava versions we were not able to use one of our method due to the version issue.

File: pom.xml (hunk 3 above)
Comment (Vipinofficial11) [reply pom.xml]: reverted.

File: src/e2e-test/features/aribasource/RunTime.feature (hunk 4 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/aribasource/RunTime.feature]: Updated

File: src/e2e-test/features/aribasource/RunTimeWithMacro.feature (hunk 5 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/aribasource/RunTimeWithMacro.feature]: updated

File: src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java (hunk 6 above)
Comment (Vipinofficial11) [reply src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java]: Renamed

File: src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java (hunk 7 above)
Comment (Vipinofficial11) [reply src/e2e-test/java/io/cdap/plugin/aribasource/stepsdesign/DesignTimeSteps.java]: Updated.

**Comment by Vipinofficial11** [comment]
> There is only one success scenario with data validation with system type `prod`. Do we expect to add other scenarios in future PRs??

Yes Once this PR is merged will add some more Run Time Scenarios, due to some of the Issues in Ariba System we were not able to successfully run the build as some tests were failing unexpectedly.

File: pom.xml (hunk 2 above)
Comment (rmstar) [reply pom.xml:279]: @Vipinofficial11 please fix

File: src/e2e-test/features/aribasource/AribaRunTime.feature [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    And Connect source as "SAP-Ariba" and sink as "BigQueryTable" to establish connection
+    And Wait with a timeout of 60000 seconds
+    And Save and Deploy Pipeline
+    And Run the Pipeline in Runtime
+    And Wait with a timeout of 960000 seconds
+    And Open and capture logs
+    And Verify the pipeline status is "Succeeded"
+    And Close the pipeline logs
+    Then Validate record created in Sink application for ViewTemplate is equal to expected output file "expectedOutputFile"
+    And Wait with a timeout of 60000 seconds
```
Comment (rmstar) [inline src/e2e-test/features/aribasource/AribaRunTime.feature]: Why do we need to wait at the end of the testcase? Also, why are sleeping for such a long time? How long does this test take to run?

File: src/e2e-test/features/aribasource/AribaRunTime.feature [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    And Enter input plugin property: "dataset" with value: "dataset"
+    And Enter input plugin property: "datasetProject" with value: "datasetprojectId"
+    And Enter input plugin property: "table" with value: "bqtarget.table"
+    And Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    And Connect source as "SAP-Ariba" and sink as "BigQueryTable" to establish connection
+    And Wait with a timeout of 60000 seconds
+    And Save and Deploy Pipeline
+    And Run the Pipeline in Runtime
+    And Wait with a timeout of 960000 seconds
```
Comment (rmstar) [inline src/e2e-test/features/aribasource/AribaRunTime.feature]: why do we need to wait between steps?

**Comment by rmstar** [comment]
> due to some of the Issues in Ariba System we were not able to successfully run the build as some tests were failing unexpectedly.

Can you provide more details on this? What tests were failing and why?

File: pom.xml (hunk 2 above)
Comment (Vipinofficial11) [reply pom.xml:279]: This dependency is actually just shifted from bottom -> top because we needed the guava version 27.0.1 but some other dependencies were using the lower versions so to override them had to move it at top.

File: src/e2e-test/features/aribasource/AribaRunTime.feature (hunk 8 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/aribasource/AribaRunTime.feature]: There is a second and minute limit to hit the API so one's the test finishes it hardly take 1 min to configure the next test and validate the plugin, so sometimes it shows null error if the minute limit is exceeded. So to avoid these kind of failures I added a 1 min wait.

Comment (Vipinofficial11) [reply src/e2e-test/features/aribasource/AribaRunTime.feature]: Screen Shot 2022-10-19 at 12 13 03 PM

File: src/e2e-test/features/aribasource/AribaRunTime.feature (hunk 9 above)
Comment (Vipinofficial11) [reply src/e2e-test/features/aribasource/AribaRunTime.feature]: This wait is actually added because the step which we had in the test framework only wait 15 min max before checking if a pipeline is succeeded or not but in Ariba sometimes pipelines are taking more than 15 mins that's why had to add a custom wait based on the observation after running few pipelines for the same test.

**Comment by Vipinofficial11** [comment]
> > due to some of the Issues in Ariba System we were not able to successfully run the build as some tests were failing unexpectedly.
> 
> Can you provide more details on this? What tests were failing and why?

**Comment by Vipinofficial11** [comment]
> > due to some of the Issues in Ariba System we were not able to successfully run the build as some tests were failing unexpectedly.
> 
> Can you provide more details on this? What tests were failing and why?

There were no specific tests which were failing there were random failures and the reason seems to be the time taken to validate and configure the pipelines.
image

**Comment by Vipinofficial11** [comment]
> > > due to some of the Issues in Ariba System we were not able to successfully run the build as some tests were failing unexpectedly.
> > 
> > 
> > Can you provide more details on this? What tests were failing and why?
> 
> There were no specific tests which were failing there were random failures and the reason seems to be the time taken to validate and configure the pipelines. image

Update :- Tried adding a test and same thing happened the build failed (dec167d295d6e2f0f1799657731ac6742f5a8838) but if I remove that new test build succeeds (fde09074717fc24a03d36c37a1c5b47268445896).

Builds : 
1. With new test : (https://github.com/data-integrations/ariba-plugins/actions/runs/3280207072) Failed
2. With Test Removed : (https://github.com/data-integrations/ariba-plugins/actions/runs/3281091324) Succeeded

**Comment by rmstar** [comment]
> > > due to some of the Issues in Ariba System we were not able to successfully run the build as some tests were failing unexpectedly.
> > 
> > 
> > Can you provide more details on this? What tests were failing and why?
> 
> There were no specific tests which were failing there were random failures and the reason seems to be the time taken to validate and configure the pipelines. image

This looks like a bug in plugin validation. Please investigate and fix.

File: src/e2e-test/features/aribasource/AribaRunTime.feature (hunk 9 above)
Comment (rmstar) [reply src/e2e-test/features/aribasource/AribaRunTime.feature]: Sleeping for 960000 seconds (> 11 days!) is too much. How long does this test take to run?

File: src/e2e-test/features/aribasource/AribaRunTime.feature [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    And Navigate to the properties page of plugin: "BigQuery"
+    And Enter input plugin property: "referenceName" with value: "Reference"
+    And Replace input plugin property: "project" with value: "projectId"
+    And Enter input plugin property: "dataset" with value: "dataset"
+    And Enter input plugin property: "datasetProject" with value: "datasetprojectId"
+    And Enter input plugin property: "table" with value: "bqtarget.table"
+    And Validate "BigQuery" plugin properties
+    And Close the Plugin Properties page
+    And Connect source as "SAP-Ariba" and sink as "BigQueryTable" to establish connection
+    And Wait with a timeout of 60000 seconds
```
Comment (rmstar) [inline src/e2e-test/features/aribasource/AribaRunTime.feature]: why are we waiting 60000 seconds (> 16 hours) here??

File: src/e2e-test/features/aribasource/AribaRunTime.feature [hunk 11]
```diff
@@ -31,7 +31,7 @@ Feature: Ariba Source - Run time scenarios
     And Enter input plugin property: "clientId" with value: "admin.clientid" for Credentials and Authorization related fields
     And Enter input plugin property: "clientSecret" with value: "admin.clientsecret" for Credentials and Authorization related fields
     And Enter input plugin property: "apiKey" with value: "admin.apiKey" for Credentials and Authorization related fields
-    And Wait with a timeout of 60000 seconds
+    And Wait with a timeout of 60000 milli seconds
```
Comment (rmstar) [inline src/e2e-test/features/aribasource/AribaRunTime.feature]: I understand you need a longer timeout for pipeline runs. However, you have added sleeps in other places like here - after entering a config property, before validating config. These extra sleeps seem unnecessary. Can we remove them?
